contracts/VToken.sol

```diff
@@ -818,6 +1093,82 @@ contract VToken is Ownable2StepUpgradeable, VTokenInterface, ExponentialNoError,
+    function swapBorrowRateMode(uint256 rateMode) external {
+        accrueInterest();
+
+        /* Verify market's block number equals current block number */
+        if (accrualBlockNumber != _getBlockNumber()) {
+            revert SwapBorrowRateModeFreshnessCheck();
+        }
+
+        address account = msg.sender;
+        uint256 variableDebt = _borrowBalanceStored(account);
+        uint256 stableDebt = _updateUserStableBorrowBalance(account);
+        uint256 accountBorrowsNew = variableDebt + stableDebt;
+        uint256 stableBorrowsNew;
+
+        if (InterestRateMode(rateMode) == InterestRateMode.STABLE) {
+            require(variableDebt > 0, "vToken: swapBorrowRateMode variable debt is 0");
+
+            stableBorrowsNew = stableBorrows + variableDebt;
+            uint256 stabelBorrowRate = stableBorrowRatePerBlock();
+
+            uint256 averageStableBorrowRateNew = ((stableBorrows * averageStableBorrowRate) +
+                (variableDebt * stabelBorrowRate)) / stableBorrowsNew;
+
+            uint256 stableRateMantissaNew = ((stableDebt * accountStableBorrows[account].stableRateMantissa) +
+                (variableDebt * stabelBorrowRate)) / accountBorrowsNew;
+
+            /////////////////////////
+            // EFFECTS & INTERACTIONS
+            // (No safe failures beyond this point)
+
+            accountStableBorrows[account].principal = stableDebt + variableDebt;
+            accountStableBorrows[account].interestIndex = stableBorrowIndex;
+            accountStableBorrows[account].stableRateMantissa = stableRateMantissaNew;
+
+            averageStableBorrowRate = averageStableBorrowRateNew;
+
+            accountBorrows[account].principal = 0;
+            accountBorrows[account].interestIndex = borrowIndex;
+        } else {
+            require(stableDebt > 0, "vToken: swapBorrowRateMode stable debt is 0");
+
+            stableBorrowsNew = stableBorrows - stableDebt;
+
+            uint256 stableRateMantissa = accountStableBorrows[account].stableRateMantissa;
+            uint256 averageStableBorrowRateNew;
+            if (stableBorrowsNew == 0) {
+                averageStableBorrowRateNew = 0;
+            } else {
+                unchecked {
+                    averageStableBorrowRateNew =
+                        ((stableBorrows * averageStableBorrowRate) - (stableDebt * stableRateMantissa)) /
+                        stableBorrowsNew;
+                }
+            }
+
+            /////////////////////////
+            // EFFECTS & INTERACTIONS
+            // (No safe failures beyond this point)
+            accountBorrows[account].principal = accountBorrowsNew;
+            accountBorrows[account].interestIndex = borrowIndex;
+
+            accountStableBorrows[account].principal = 0;
+            accountStableBorrows[account].interestIndex = stableBorrowIndex;
+
+            averageStableBorrowRate = averageStableBorrowRateNew;
+        }
+
+        stableBorrows = stableBorrowsNew;
+
```

Comment:
We could also move here `averageStableBorrowRate = averageStableBorrowRateNew;`

Reply:
Fixed, can verify the changes at [commit](https://github.com/VenusProtocol/isolated-pools/pull/132/commits/0f14951544955613711dd9321a85af4f96073049).